pkg/image/nbdkit.go

```diff
@@ -208,16 +219,20 @@ func (n *Nbdkit) StartNbdkit(source string) error {
 	}
 	n.c.Stderr = n.c.Stdout
 	output := bufio.NewReader(stdout)
-	go func() {
-		for {
-			line, err := output.ReadString('\n')
-			if err != nil {
-				break
+	if n.LogWatcher != nil {
+		n.LogWatcher.Start(output)
+	} else {
+		go func() {
+			for {
+				line, err := output.ReadString('\n')
```

Comment:
You could use the scanner here as well. Not hugely passionate about it, if you need this in sooner rather than later, I won't block it

Reply:
Sure, I can change this over too. I don't think this feature is time sensitive.